ghci: add page (#4576)
tldr-pages/tldr: pages/common/ghci.md

# ghci

> The Glasgow Haskell Compiler's interactive environment.
> More information: <https://downloads.haskell.org/ghc/latest/docs/html/users_guide/ghci.html>.

> - Start an interactive REPL session:

Reviewer: Based on the `node.md` page:
```suggestion
- Start a REPL (interactive shell):
```

Author: That's true, sounds better. I will replace all instances of "an interactive REPS session" with "start a REPL (interactive shell)"

`ghci`

- Start a REPL and load the specified Haskell source file:

`ghci {{source_file.hs}}`

- Start a REPL and enable a language option:

`ghci -X{{language_option}}`

> - Start a REPL and enable almost all compiler warnings:

Reviewer: `enabled` or `turned on` should be in past tense I think, because you start a REPL with an option turned on. Maybe:
```suggestion
- Start a REPL with compiler warnings that indicate potentially suspicious code:
```
Or
```suggestion
- Start a REPL with compiler warnings that indicate potentially suspicious code turned on:
```
Sorry, English is not my mother language cc @sbrl @owenvoke

Author: I wouldn't say "potentially suspicious" to be honest. Something about it sounds ...off. I changed the wording and the code-snippet a bit.

Reviewer: I suggested that because that's the official language in the docs.

Author: Maybe my non-native ears are not qualified to rate the wording of english sentences then :D I will add the phrase.

Reviewer: I'm from Argentina. The Oxford dictionary here is @sbrl :hand_over_mouth:.

Reviewer 2: Lol! In this case, it really didn't need changing :P

It's fine as-is, so no need to adjust again, but in future it didn't need adjusting :laughing:

> - Start a REPL and enable some level of compiler warnings to indicate potentially suspicious code:
> 
> `ghci -W{{all|compat|...}}`

Reviewer: Is `-Wall` the most used option? Otherwise you can use:
```suggestion
- Start a REPL and enable some level of compiler warnings (e.g. `all` or `compact`):

`ghci -W{{warning_level}}`
```
If you the pipe here: `{{all|compact|...}}` you should list all the available options, but there are a lot of them. Or, revert to `ghci -W{{all}}`

Author: I think that `-Wall` is the most used one, could be wrong though. I would still go with your suggestion if that's ok. It seems to be the overall better choice.

> `ghci -Wall`

Reviewer: I can't verify this argument, even with the command installed. The official man page is not friendly to me.

Author: I verified it by starting `ghci -Wall` and evaluating the expression `2*2`. This immediately triggers a warning which would not be there without `-Wall`.

Reviewer: @gutjuri Oh cool, and How can I test the `-X` example then?

Author: For example, start `ghci -XNoImplicitPrelude` and try calling something from the Prelude (e.g. `take 5 [1..]`). 

This should also work with any other language option, though many of them are not easily testable in a REPL in my opinion.

Reviewer: Oh `all` is an argument option. Based on [sonar-scanner](https://github.com/tldr-pages/tldr/blob/master/pages/common/sonar-scanner.md):
```suggestion
`ghci -W{{all}}`
```

> - Start an interactive REPL session with the specified search path for finding source files:

Reviewer: is `dir` a `directory`? Is this better?
```suggestion
- Start an interactive REPL session with a colon-separated list of directories for finding source files:
```

Author: yeah, thanks

> `ghci -i{{dir1:...:dirn}}`

Reviewer: I think this is better if I can use a relative path to a directory:
```suggestion
`ghci -i{{path/to/directory1}}:{{path/to/directory2}}`
```
Or if the directories should be in the local path:
```suggestion
`ghci -i{{directory1}}:{{directory2}}`
```

Author: Relative paths can indeed be used. I will incorporate your suggestion.
